### PR TITLE
Add $HOSTNAME and $HOSTTYPE environment variables

### PR DESCRIPTION
**Pull request by laloch**
Adds `$HOSTNAME` and `$HOSTTYPE` variables that the user might expect to exist.

`$HOSTNAME` defaults to `platform.uname().node`.
`$HOSTTYPE` defaults to `platform.uname().machine`.

### CODE REVIEW

**Comment by laloch** [comment]
OK, there's no `os.uname()` on Windows :roll_eyes:

File: xonsh/environ.py [hunk 1]
```diff
@@ -87,6 +87,7 @@
 )
 import xonsh.prompt.base as prompt
 
+from platform import uname
```
Comment (scopatz) [inline xonsh/environ.py:90]: Please `import platform` for amalgamation

File: xonsh/environ.py [hunk 2]
```diff
@@ -942,6 +943,20 @@ def DEFAULT_VARS():
             "Note: ``erasedups`` is supported only in sqlite backend).",
             doc_store_as_str=True,
         ),
+        "HOSTNAME": Var(
+            is_string,
+            ensure_string,
+            ensure_string,
+            uname().node,
```
Comment (scopatz) [inline xonsh/environ.py:950]: This should be something like: `default_value(lambda env: uname.node)` so it is not called when the deafults are created.

File: xonsh/environ.py [hunk 3]
```diff
@@ -942,6 +943,20 @@ def DEFAULT_VARS():
...
+            "Automatically set to the name of the current host.",
+        ),
+        "HOSTTYPE": Var(
+            is_string,
+            ensure_string,
+            ensure_string,
+            uname().machine,
```
Comment (scopatz) [inline xonsh/environ.py:957]: Similarly, this should be `default_value(lambda env: uname.machine)`

File: xonsh/environ.py (hunk 1 above)
Comment (laloch) [reply xonsh/environ.py:90]: Oops, sorry!

File: xonsh/environ.py (hunk 2 above)
Comment (laloch) [reply xonsh/environ.py:950]: I actually wanted it to be called only once.

**Comment by scopatz** [comment]
Thanks for putting this in @laloch!

File: xonsh/environ.py (hunk 1 above)
Comment (laloch) [reply xonsh/environ.py]: Done.

File: xonsh/environ.py (hunk 2 above)
Comment (laloch) [reply xonsh/environ.py]: Done.

Comment (laloch) [reply xonsh/environ.py]: Done.

File: xonsh/environ.py (hunk 3 above)
Comment (laloch) [reply xonsh/environ.py]: Done.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**